### PR TITLE
[DCOS-41593][Integration Tests] Add Mesos Secrets support to the spark dispatcher

### PR DESCRIPTION
**Pull request by farhan5900**
## What changes were proposed in this pull request?

Resolves [DCOS-41593](https://jira.mesosphere.com/browse/DCOS-41593)

* Adds multiple integration tests for various configurations of Mesos Secrets support.

## How were these changes tested?

It is tested manually submitting jobs.

## Release Notes

Adds Integration Tests for Mesos Secrets

### CODE REVIEW

File: tests/test_spark_secrets.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    secret_path = "/spark/secret-name"
+    secret_value = "secret-value"
+    dcos_utils.delete_secret(secret_path)
+    dcos_utils.create_secret(secret_path, secret_value, False)
+    try:
+        utils.run_tests(
+            app_url=utils.dcos_test_jar_url(),
+            app_args=auth_token,
+            expected_output=secret_value,
+            args=["--conf=spark.mesos.driver.secret.names='"+secret_path+"'",
```
Comment (akirillov) [inline tests/test_spark_secrets.py]: let's use `"--conf=spark.mesos.driver.secret.names='{}'".format(secret_path)` instead

File: tests/test_spark_secrets.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                  "--conf=spark.mesos.driver.secret.envkeys='SECRET_ENV_KEY'",
```
Comment (akirillov) [inline tests/test_spark_secrets.py]: do we need single quotes around config values? if not - let's remove them

File: tests/test_spark_secrets.py (hunk 1 above)
Comment (farhan5900) [reply tests/test_spark_secrets.py]: Changed it to using format method.

File: tests/test_spark_secrets.py (hunk 2 above)
Comment (farhan5900) [reply tests/test_spark_secrets.py]: Removed single quotes.

**Pull request merged**